pkg/zfs/zfs_util.go

```diff
@@ -216,15 +222,21 @@ func buildDatasetCreateArgs(vol *apis.ZFSVolume) []string {
 	ZFSVolArg = append(ZFSVolArg, ZFSCreateArg)
 
 	if len(vol.Spec.Capacity) != 0 {
-		quotaProperty := "quota=" + vol.Spec.Capacity
+		quotaProperty := vol.Spec.QuotaType + "=" + vol.Spec.Capacity
 		ZFSVolArg = append(ZFSVolArg, "-o", quotaProperty)
 	}
 	if len(vol.Spec.RecordSize) != 0 {
 		recordsizeProperty := "recordsize=" + vol.Spec.RecordSize
 		ZFSVolArg = append(ZFSVolArg, "-o", recordsizeProperty)
 	}
 	if vol.Spec.ThinProvision == "no" {
-		reservationProperty := "reservation=" + vol.Spec.Capacity
+		reservationProperty := ""
+		switch vol.Spec.QuotaType {
+		case "quota":
+			reservationProperty = "reservation=" + vol.Spec.Capacity
+		case "refquota":
+			reservationProperty = "refreservation=" + vol.Spec.Capacity
+		}
```

Comment:
How about simplifying this by using a method and calling it here. That way we only need to modify the method to change something and not at all places?

```
// get the reservation property based on the quota type
func reservationProperty(quotaType string, capacity string) string {
	var reservationProperties = map[string]string{
		"quota":    "reservation=",
		"refquota": "refreservation=",
	}
	return reservationProperties[quotaType] + capacity
}
```

```
ZFSVolArg = append(ZFSVolArg, "-o", reservationProperty(vol.Spec.QuotaType, vol.Spec.Capacity))
```